gpustack/migrations/versions/2024_12_26_1409-e6bf9e067296_add_model_categories.py

```diff
@@ -106,7 +106,133 @@ def upgrade() -> None:
+                WHERE TABLE_NAME = 'model_instances' 
+                AND COLUMN_NAME = 'state' 
+                AND TABLE_SCHEMA = DATABASE()
+            """)
+        ).scalar()
+
+        existing_values = []
+        if result:
+            enum_str = result.split("enum(")[1].split(")")[0]
+            existing_values = [v.strip("'") for v in enum_str.split("','")]
+
+        # 添加新值（STARTING 和 UNREACHABLE）
+        new_values = existing_values + ['STARTING', 'UNREACHABLE']
+        new_enum_str = "enum('" + "','".join(new_values) + "')"
+
+        # 构造新的 ALTER TABLE 语句
+        alter_sql = f"""
+            ALTER TABLE model_instances 
+            MODIFY COLUMN state {new_enum_str};
+        """
+
+        # 执行修改
+        conn.execute(sa.text(alter_sql))
+
+        # 获取现有 workerstateenum 的值
+        result = conn.execute(
+            sa.text("""
+                SELECT COLUMN_TYPE 
+                FROM information_schema.COLUMNS 
+                WHERE TABLE_NAME = 'workers' 
+                AND COLUMN_NAME = 'state' 
+                AND TABLE_SCHEMA = DATABASE()
+            """)
+        ).scalar()
+
+        existing_values = []
+        if result:
+            enum_str = result.split("enum(")[1].split(")")[0]
+            existing_values = [v.strip("'") for v in enum_str.split("','")]
+
+        # 添加新值（UNREACHABLE）
+        if 'UNREACHABLE' not in existing_values:
+            new_values = existing_values + ['UNREACHABLE']
+            new_enum_str = "enum('" + "','".join(new_values) + "')"
+
+            # 构造新的 ALTER TABLE 语句
+            alter_sql = f"""
+                ALTER TABLE workers 
+                MODIFY COLUMN state {new_enum_str};
+            """
+
+            # 执行修改
+            conn.execute(sa.text(alter_sql))
+
+        # 扩展model_usage_operation_enum 类型
+        # 获取现有 operationenum 的值
+        result = conn.execute(
+            sa.text("""
+                SELECT COLUMN_TYPE 
+                FROM information_schema.COLUMNS 
+                WHERE TABLE_NAME = 'model_usages' 
+                AND COLUMN_NAME = 'operation' 
+                AND TABLE_SCHEMA = DATABASE()
+            """)
+        ).scalar()
+
+        existing_values = []
+        if result:
+            enum_str = result.split("enum(")[1].split(")")[0]
+            existing_values = [v.strip("'") for v in enum_str.split("','")]
+
+        # 添加新值（AUDIO_TRANSCRIPTION 和 AUDIO_SPEECH）
+        new_values = existing_values.copy()
+        for value in ['AUDIO_TRANSCRIPTION', 'AUDIO_SPEECH']:
+            if value not in existing_values:
+                new_values.append(value)
+
+        new_enum_str = "enum('" + "','".join(new_values) + "')"
 
+        # 构造新的 ALTER TABLE 语句
+        alter_sql = f"""
+            ALTER TABLE model_usages 
+            MODIFY COLUMN operation {new_enum_str};
+        """
+
+        # 执行修改
+        conn.execute(sa.text(alter_sql))
+        
+        # MySQL的workers status更新部分
```

Comment:
Can you translate the comments to English

Reply:
ok